src/tools/genpolicy/src/policy.rs

```diff
@@ -0,0 +1,900 @@
+                Hooks: None,
+                Annotations: annotations,
+                Linux: linux,
+            },
+            storages,
+            exec_commands,
+        }
+    }
+
+    fn get_container_process(
+        &self,
+        resource: &dyn yaml::K8sResource,
+        yaml_container: &pod::Container,
+        is_pause_container: bool,
+        namespace: &str,
+        c_settings: &KataSpec,
+        is_privileged: bool,
+    ) -> KataProcess {
+        // Start with the Default Unix Spec from
+        // https://github.com/containerd/containerd/blob/release/1.6/oci/spec.go#L132
+        let mut process = containerd::get_process(is_privileged, &self.settings.common);
+
+        yaml_container.apply_capabilities(&mut process.Capabilities, &self.settings.common);
+
+        let (yaml_has_command, yaml_has_args) = yaml_container.get_process_args(&mut process.Args);
+        yaml_container
+            .registry
+            .get_process(&mut process, yaml_has_command, yaml_has_args);
+
+        if let Some(tty) = yaml_container.tty {
+            process.Terminal = tty;
+            if tty && !is_pause_container {
+                process.Env.push("TERM=xterm".to_string());
+            }
+        }
+
+        if !is_pause_container {
+            process.Env.push("HOSTNAME=$(host-name)".to_string());
+        }
+
+        let service_account_name = if let Some(s) = &yaml_container.serviceAccountName {
+            s
+        } else {
+            "default"
+        };
+
+        yaml_container.get_env_variables(
+            &mut process.Env,
+            &self.config_maps,
+            &self.secrets,
+            namespace,
+            resource.get_annotations(),
+            service_account_name,
+        );
+
+        substitute_env_variables(&mut process.Env);
+        substitute_args_env_variables(&mut process.Args, &process.Env);
+        c_settings.get_process_fields(&mut process);
+        process.NoNewPrivileges = !yaml_container.allow_privilege_escalation();
+
+        process
+    }
+}
+
+impl KataSpec {
+    fn add_annotations(&self, annotations: &mut BTreeMap<String, String>) {
+        for a in &self.Annotations {
+            annotations.entry(a.0.clone()).or_insert(a.1.clone());
+        }
+    }
+
+    fn get_process_fields(&self, process: &mut KataProcess) {
+        if process.User.UID == 0 {
+            process.User.UID = self.Process.User.UID;
+        }
+        if process.User.GID == 0 {
+            process.User.GID = self.Process.User.GID;
+        }
+
+        process.User.AdditionalGids = self.Process.User.AdditionalGids.to_vec();
+        process.User.Username = String::from(&self.Process.User.Username);
+        add_missing_strings(&self.Process.Args, &mut process.Args);
+
+        add_missing_strings(&self.Process.Env, &mut process.Env);
+    }
+}
+
+fn get_image_layer_storages(
+    storages: &mut Vec<agent::Storage>,
+    image_layers: &Vec<registry::ImageLayer>,
+    root: &KataRoot,
+) {
```

Comment:
Hey @danmihai1. Am I correct in assuming that this is an interface for sharing image layers? I am thrilled to see this feature, as I am working on integrating image sharing feature into the main branch (https://github.com/kata-containers/kata-containers/pull/7837). I would like to pass the dm-verity information of the image or layer to the guest via the policy to check the info from host is correct as expected, if possible. I would appreciate your assistance with this task.

Reply:
Please ping me on Slack when you are ready to discuss this. 

Yes, the dm-verity information of each container layer is being verified already when using the Policy generated by the genpolicy tool. But, the current Policy is based on Wedson's snapshotter design/implementation that we discussed several months ago. Hopefully there aren't big differences between your version and Wedson's. If there are significant differences, we'll have to modify this tool to support both.